### PR TITLE
Add msquic client support to QNS

### PR DESCRIPTION
**Pull request by anrossi**

### CODE REVIEW

**Comment by nibanks** [comment]
FYI, quant's client interop file: https://github.com/NTAP/quant/blob/master/test/interop.sh#L25. It should give you an idea of the args passed to the client.

File: .azure/templates/run-qns.yml [hunk 1]
```diff
@@ -39,6 +39,12 @@ jobs:
             python3 pull.py -i ${{ client }},${{ server }}
           displayName: Setup Runner
 
+        ${{ if ne(parameters.build, '') }}:
+          - script: |
+              sed 's/msquic\/qns:latest/msquic\/qns:custom/' implementations.json > modified.json
```
Comment (nibanks) [inline .azure/templates/run-qns.yml:44]: Must happen before `pull.py` above

File: .azure/templates/run-qns.yml [hunk 2]
```diff
@@ -39,6 +39,12 @@ jobs:
             python3 pull.py -i ${{ client }},${{ server }}
           displayName: Setup Runner
 
+        ${{ if ne(parameters.build, '') }}:
```
Comment (nibanks) [inline .azure/templates/run-qns.yml:42]: You need to default `parameters.build` to `''` above

File: .azure/templates/run-qns.yml [hunk 3]
```diff
@@ -36,9 +37,18 @@ jobs:
             git clone "https://github.com/marten-seemann/quic-interop-runner.git"
             cd quic-interop-runner
             pip3 install -r requirements.txt
-            python3 pull.py -i ${{ client }},${{ server }}
           displayName: Setup Runner
 
+        ${{ if ne(parameters.build, '') }}:
```
Comment (nibanks) [inline .azure/templates/run-qns.yml:42]:
```suggestion
        - ${{ if ne(parameters.build, '') }}:
```

File: .azure/templates/run-qns.yml [hunk 4]
```diff
@@ -36,9 +37,18 @@ jobs:
...
           displayName: Setup Runner
 
+        - ${{ if ne(parameters.build, '') }}:
+          - script: |
+              sed 's/msquic\/qns:latest/msquic\/qns:custom/' implementations.json > modified.json
+              mv modified.json implementations.json
+            displayName: Replace msquic image
+
+        - script: |
+            python3 pull.py -i ${{ client }},${{ server }}
```
Comment (nibanks) [inline .azure/templates/run-qns.yml:49]:
```suggestion
            cd quic-interop-runner
            python3 pull.py -i ${{ client }},${{ server }}
```

File: .azure/templates/run-qns.yml [hunk 5]
```diff
@@ -36,9 +37,21 @@ jobs:
             git clone "https://github.com/marten-seemann/quic-interop-runner.git"
             cd quic-interop-runner
             pip3 install -r requirements.txt
-            python3 pull.py -i ${{ client }},${{ server }}
           displayName: Setup Runner
 
+        - ${{ if ne(parameters.build, '') }}:
+          - script: |
+              cd quic-interop-runner
+              TAG={{ parameters.build }}
```
Comment (nibanks) [inline .azure/templates/run-qns.yml:45]:
```suggestion
              TAG=${{ parameters.build }}
```

**Comment by larseggert** [comment]
I think you need to also change this line to activate the client tests: https://github.com/marten-seemann/quic-interop-runner/blob/master/implementations.json#L48

**Review by nibanks — APPROVED** [review]

**Pull request merged**